py/objstr.c

```diff
@@ -743,11 +743,27 @@ STATIC mp_obj_t str_finder(size_t n_args, const mp_obj_t *args, int direction, b
     const mp_obj_type_t *self_type = mp_obj_get_type(args[0]);
     check_is_str_or_bytes(args[0]);
 
-    // check argument type
-    str_check_arg_type(self_type, args[1]);
-
     GET_STR_DATA_LEN(args[0], haystack, haystack_len);
-    GET_STR_DATA_LEN(args[1], needle, needle_len);
+
+    byte needle_data;
+    const byte *needle;
+    size_t needle_len;
+    if (mp_obj_is_int(args[1]) && (self_type == &mp_type_bytes
```

Comment:
Would it reduce code size if we used `mp_obj_get_int_maybe()` instead of calling `mp_obj_is_int()` and `mp_obj_get_int()` separately?

Reply:
Even if not CPython compliant, would it be so bad if we left out the `self_type` checks to save a few more bytes?

Reply:
I'm assuming though it wouldn't really make sense to compare an integer representation of a single byte to a `str` instance?

Rather than checking against byte and bytearray, would it be simpler to just check !=  str?

Reply:
@andrewleech If you mean
```py
"abc".find(97)
```
that's now allowed in CPython. `find(int)` only allowed on bytes/bytearray.

But yes I think you're right about just comparing to str. Saves 8 bytes :D

Reply:
> Would it reduce code size if we used mp_obj_get_int_maybe() instead of calling mp_obj_is_int() and mp_obj_get_int() separately?

Yes! Saves 16 bytes :partying_face: 

> Even if not CPython compliant, would it be so bad if we left out the `self_type` checks to save a few more bytes?

I assume you mean the `check_is_str_or_bytes(args[0]);` line that's in most of these functions. I also wondered this on Friday but they are actually no-ops on the default build. This check isn't normally needed because self must be the right type (because this method came from teh type's locals dict) but you can also write
```
x = str.index
x(int, 'a')
```

So  `check_is_str_or_bytes` is verifying the self argument in the case where it's being used as an unbound method. However calls `mp_check_self` which is nothing on builds with `MICROPY_BUILTIN_METHOD_CHECK_SELF_ARG` enabled (because in that case `mp_convert_member_lookup` makes a special checked function to verify this).